toolkit/tools/pkg/imagecustomizerlib/extractpartitions_test.go

```diff
@@ -0,0 +1,142 @@
+package imagecustomizerlib
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/file"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/shell"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAddSkippableFrame(t *testing.T) {
+	// Create a skippable frame containing the metadata and prepend the frame to the partition file
+	skippableFrameMetadata, err := createSkippableFrameMetadata()
+	assert.NoError(t, err)
+
+	// Create test raw partition file
+	partitionRawFilepath, err := createTestRawPartitionFile("test.raw")
+	assert.NoError(t, err)
+
+	// Compress to .raw.zst partition file
+	partitionFilepath, err := compressWithZstd(partitionRawFilepath)
+	assert.NoError(t, err)
+
+	// Test adding the skippable frame
+	err = addSkippableFrame(partitionFilepath, SkippableFrameMagicNumber, SkippableFrameSize, skippableFrameMetadata)
+	assert.NoError(t, err)
+
+	// Verify decompression with skippable frame
+	err = verifySkippableFrameDecompression(partitionRawFilepath, partitionFilepath)
+	assert.NoError(t, err)
+
+	// Verify skippable frame metadata
+	err = verifySkippableFrameMetadataFromFile(partitionFilepath, SkippableFrameMagicNumber, SkippableFrameSize, skippableFrameMetadata)
+	assert.NoError(t, err)
+
+	// Remove test partition files
+	err = os.Remove(partitionRawFilepath)
+	assert.NoError(t, err)
+
+	err = os.Remove(partitionFilepath)
+	assert.NoError(t, err)
+}
+
+func createTestRawPartitionFile(filename string) (string, error) {
+	// Test data
+	data := []byte{0x01, 0x02, 0x03, 0x04, 0x05}
+
+	err := writeToFile(filename, data)
+	if err != nil {
+		return "", err
+	} else {
+		logger.Log.Infof("File %s created successfully with dummy data.", filename)
```

Comment:
nit: follow this format -
logger.Log.Infof("Partition file created: %s", partitionFilepath)